Set-DomainUser.ps1

```diff
@@ -1,720 +1,1124 @@
+            If ($Null -ne $regPath) {
+                $retInstallPath = $regPath.ImagePath.Substring($regPath.ImagePath.IndexOf('"'), $regPath.ImagePath.LastIndexOf('"') + 1)
+            }
+        }
+        catch {
+            Throw $(New-Object System.Exception ("Cannot get Service Install path for $ServiceName", $_.Exception))
+        }
+        
+        return $retInstallPath
+    }
+    End {
+        
+    }
+}
+
+Function New-ConnectionToRestAPI {
+    <#
+    .SYNOPSIS
+    Get the installation path of a service
+    .DESCRIPTION
+    The function receive the service name and return the path or returns NULL if not found
+    .EXAMPLE
+    (Get-ServiceInstallPath $<ServiceName>) -ne $NULL
+    .PARAMETER pvwaAddress
+    The PVWA server address (e.g. https://subdomain.privilegecloud.cyberark.cloud)
+    .PARAMETER tinaCreds
+    Tenant administrator/installer user credentials
+    #>
+    # Get PVWA and login informatioN
+    param (
+        [Parameter(Mandatory = $true)]
+        $pvwaAddress,
+        [Parameter(Mandatory = $true)]
+        [PSCredential]$tinaCreds        
+    )
+    $url = $pvwaAddress + "/PasswordVault/API/auth/Cyberark/Logon"
+    $BSTR = [System.Runtime.InteropServices.Marshal]::SecureStringToBSTR($tinaCreds.Password)
+    
+    $headerPass = [System.Runtime.InteropServices.Marshal]::PtrToStringAuto($BSTR)
+    $body = @{
+        username = $tinaCreds.UserName
+        password = $headerPass
+    }
+    $json = $body | ConvertTo-Json
+    Try {
+        $pvwaToken = Invoke-RestMethod -Method 'Post' -Uri $url -Body $json -ContentType 'application/json'
+    }
+    Catch {
+        Write-Host "Failed to retrieve token. Response received:"
+        Write-Host $_.Exception.Message
+        exit 1
+    }
+    if ($pvwaToken -match "[0-9a-zA-Z]{200,256}") {
+        return $pvwaToken
+    }
+    else {
+        Write-Host "Failed to retrieve token. Response received:"
+        Write-Host $_.Exception.Message
+        exit 1
+    }
+}
+
+Function Test-PvwaToken {
+    <#
+    .SYNOPSIS
+    Test a PVWA token to ensure it is valid
+    .DESCRIPTION
+    The function receive the service name and return the path or returns NULL if not found
+    .EXAMPLE
+    Test-PvwaToken -Token $Token -PvwaAddress https://subdomain.privilegecloud.cyberark.cloud
+    .PARAMETER pvwaAddress
+    The PVWA server address (e.g. https://subdomain.privilegecloud.cyberark.cloud)
+    .PARAMETER Token
+    PVWA Token
+    #>
+    param (
+        [Parameter(Mandatory = $true)]
+        [string]$pvwaAddress,
+        [Parameter(Mandatory = $true)]
+        [string]$Token        
+    )
+    $url = $pvwaAddress + "/PasswordVault/API/Accounts?limit=1"
+    $Headers = @{
+        Authorization = $Token
+    }
+    $testToken = Invoke-RestMethod -Method 'Get' -Uri $url -Headers $Headers -ContentType 'application/json'
+    if ($testToken) {
+        return $true
+    }
+    else {
+        return $false
+    }
+}
```

Comment:
Function to test PVWA token by attempting to retrieve a list of accounts (this will be successful even if there are no accounts available)